app/main/lib/image_similarity.py

```diff
@@ -47,11 +49,30 @@ def save(image):
     db.session.rollback()
     raise e
 
+
+@tenacity.retry(wait=tenacity.wait_fixed(0.5), stop=tenacity.stop_after_delay(5), after=_after_log)
+def alter_pdq(image):
+  try:
+    # First locate existing image and append new context
+    existing = db.session.query(ImageModel).filter(ImageModel.url==image.url).one()
```

Comment:
We should be looking up based on the id, not URL. Remove the code around image.context (we don't want to alter that).